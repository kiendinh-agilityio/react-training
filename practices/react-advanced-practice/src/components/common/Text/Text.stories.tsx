import Text from '.';

export default {
  component: Text,
  title: 'Text',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const TextContent = {
  args: {
    content:
      'Hi, I’m Alec Thompson, Decisions: If you can’t decide, the answer is no. If two equally difficult paths, choose the one more painful in the short term (pain avoidance is creating an illusion of equality).',
  },
};
