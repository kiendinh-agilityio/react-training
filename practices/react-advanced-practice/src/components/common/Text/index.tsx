// Import radix ui
import { Text as TextBase, TextProps } from '@radix-ui/themes';

// Import types
import { TextSize } from '@/types';

type TextBaseProps = {
  children: React.ReactNode;
  size?: TextSize;
} & TextProps;

const Text = ({
  children,
  size = TextSize.Small,
  as = 'p',
  weight = 'regular',
  className = '',
}: TextBaseProps) => (
  <TextBase as={as} weight={weight} className={`${size} ${className}`}>
    {children}
  </TextBase>
);

export default Text;
