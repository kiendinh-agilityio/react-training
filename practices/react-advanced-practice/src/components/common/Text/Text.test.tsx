import { render } from '@testing-library/react';

// Component
import Text from '.';

describe('Text Component', () => {
  it('should match with snapshot', () => {
    const { container } = render(
      <Text content="This is a text" className="font-regular" />,
    );

    expect(container).toMatchSnapshot();
  });
});
