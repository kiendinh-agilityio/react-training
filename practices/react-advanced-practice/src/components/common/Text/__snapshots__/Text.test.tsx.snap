// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should match with snapshot 1`] = `
<div>
  <p
    class="rt-Text rt-r-weight-regular text-sm font-regular font-regular"
  >
    This is a text
  </p>
</div>
`;
